### ISSUE TITLE
Low Precision and AUC on Hybrid model than CF

### ISSUE THREAD

**Issue by rajat1602** [lyst/lightfm#297]
Hi, 
I have the following dataset:
1. User-Item Interaction of shape (15732,48947)
2.User Feature matrix (CSR) of shape (15732,26743) . It includes horizontally stacked per user Identity matrix. 
3.Item Feature matrix (CSR) of shape (48947,116457) . It includes horizontally stacked per item Identity matrix. 
Below are the hyperparametes I am using:
**model = LightFM(no_components= 30,learning_rate=0.03,loss='warp',learning_schedule='adagrad',random_state=86)
model.fit(users_items_pivot_matrix,user_features = user_features_matrix,item_features = item_features_matrix, 
          epochs=50,verbose=True)**
AUC : 0.52

When fitting model without User and Item Features, getting AUC = 0.97.
What's the issue ??

**Comment by maciejkula** [comment]
1. When adding features you may need to change hyperparameters: have you performed a hyperparameter search after adding features?
2. Are the features high quality? Are any of them not normalized? If there are features that are shared between many users or items, performing tf-idf weighting on the features matrix may be helpful.

**Comment by rajat1602** [comment]
1. Currently, doing the same.
2. Both user and item features are binary. 1 if the feature is available for that particular user or item.
So, weighting is either 0 or 1. 
Note: feature matrix has been transformed to datatye = np.float32

**Comment by maciejkula** [comment]
Can you try transforming your features so that the feature weights in every row sum to 1?

**Comment by rajat1602** [comment]
Alright! will try this too.
Do I need to normalize my interaction matrix as Interaction matrix contains ranking from 0  to 5? and Feature Matrices are binary.

**Comment by maciejkula** [comment]
For ranking losses the interaction matrix should have zeros and ones.

**Comment by rajat1602** [comment]
Ok...I have made the few changes which you suggested. Will let you know the results.
One more thing, while predicting for new user and new item do I need to pass user / item feature for that particular user and item or the whole user feature and item feature plus the new user / item feature used during training?

**Comment by rajat1602** [comment]
Hi, It worked man!!....Thanks much...Getting 
AUC: 0.98 after using both User and Item Feature.

One more thing, while predicting for new user and new item do I need to pass user / item feature for that particular user and item or the whole user feature and item feature plus the new user / item feature used during training?

**Comment by maciejkula** [comment]
Great!

You will most likely want to pass the entire matrix: user/item indices index into its rows.

**Comment by maciejkula** [comment]
Sure, if you want to predict for items `0..19`, you just need the first 20 rows of the matrix. But what if you want to predict for items `[0, 14000, 45000]`?

**Comment by rajat1602** [comment]
Let me tell you my business problem. I am making recommendation for Market Research. So, I get almost 100-150 new items daily. So there are no interactions for these. I just only want to recommend these items in periodic fashion (say 1 hour) to the existing user in database.
1. Do I need to make the whole new interaction matrix daily, which means suppose U1 interact with 1 item from 100 items, so i can use this interaction to recommend for different users which may have same interaction pattern from last 2 day interaction matrix for those items.
2. Or, just build the model once with huge dataset say, past 1 year data and train the interaction / user and item features...pickle the model and just pass the ids and features.

For above query:
No, these 20 items are new items. So there are no interactions for these item in user-item matrix. and that User has suppose index of 1000 in interaction as well as user feature matrix and want to predict for these 20 items for this user. Do I need to pass the index 1000 in predict or 0 plus what will be the user / feature matrix? Do I need to append the user / feature matrix of these 20 new items in the user / feature matrix used during training ??

Any help will be much appreciated!!

**Comment by maciejkula** [comment]
(This issue may be helpful https://github.com/lyst/lightfm/issues/298)

**Comment by SimonCW** [comment]
I’m closing this issue because it has been inactive for a long time. If you still encounter the problem, please open a new issue.

Thank you!
